ui/src/warp_runner/manager/commands/constellation_commands.rs

```diff
@@ -64,3 +88,184 @@ fn initialize_items(warp_storage: &mut warp_storage) -> Result<uplink_storage, E
+            .await
+        {
+            Ok(mut upload_progress) => {
+                let mut previous_percentage: usize = 0;
+                let mut upload_process_started = false;
+
+                while let Some(upload_progress) = upload_progress.next().await {
+                    match upload_progress {
+                        Progression::CurrentProgress {
+                            name,
+                            current,
+                            total,
+                        } => {
+                            if !upload_process_started {
+                                upload_process_started = true;
+                                log::info!("Starting upload for {name}");
+                                log::info!("0% completed -> written 0 bytes")
+                            };
+
+                            if let Some(total) = total {
+                                let current_percentage =
+                                    (((current as f64) / (total as f64)) * 100.) as usize;
+                                if previous_percentage != current_percentage {
+                                    previous_percentage = current_percentage;
+                                    log::info!(
+                                        "{}% completed -> written {current} bytes",
+                                        (((current as f64) / (total as f64)) * 100.) as usize
+                                    )
+                                }
+                            }
+                        }
+                        Progression::ProgressComplete { name, total } => {
+                            log::info!(
+                                "{name} has been uploaded with {} MB",
+                                total.unwrap_or_default() / 1024 / 1024
+                            );
+                        }
+                        Progression::ProgressFailed {
+                            name,
+                            last_size,
+                            error,
+                        } => {
+                            log::info!(
+                                "{name} failed to upload at {} MB due to: {}",
+                                last_size.unwrap_or_default(),
+                                error.unwrap_or_default()
+                            );
+                        }
+                    }
+                }
+                match set_thumbnail_if_file_is_image(warp_storage.clone(), filename.clone()).await {
+                    Ok(success) => log::info!("{:?}", success),
+                    Err(error) => log::error!("Error on update thumbnail: {:?}", error),
+                }
+                log::info!("{:?} file uploaded!", &filename);
+            }
+            Err(error) => log::error!("Error when upload file: {:?}", error),
+        }
+    }
+    get_items_from_current_directory(warp_storage)
+}
+
+fn verify_duplicate_name(
```

Comment:
i wonder if it's better to notify the user that the file exists and then 
- ask them if the file should be overwritten
- if yes: overwrite the file
- if no, fail the operation (let the user rename the file themself before retrying)

otherwise, if there is a file named "file.txt", the first duplicate would be "file 1.txt", next would be "file 1 2.txt", etc.